doublex/proxy.py

```diff
@@ -202,14 +202,21 @@ def get_arg_spec(self):
         del retval.args[0]
         return retval
 
+    def is_classmethod(self):
+        return (
+            inspect.ismethod(self.method) and self.method.__self__ is self.proxy.collaborator_class
+        )
+
     def get_call_args(self, context):
         args = context.args
-        if self.proxy.isclass():
+        if self.proxy.isclass() and not self.is_classmethod():
             args = (None,) + args  # self
 
         retval = getcallargs(self.method, *args, **context.kargs)
         if 'self' in retval:
```

Comment:
Should we perhaps add a check that this should not be a `classmethod` as well, to allow arguments named `self` in class methods? It's a rare case but why not?

Disclaimer: I'm reviewing on my phone.

Reply:
While I can't think of a case were `self` as an argument in a classmethod is a good idea, supporting it is cleaner than raising a `TypeError`. Will fix!

Reply:
Fixed!